### ISSUE TITLE
Bug of finetuning code? the attention mask of padding is not 0.

### ISSUE THREAD

**Issue by Tebmer** [THUDM/GLM#70]
Hi, 

I am using your finetuning code of seq2seq customization. 

However, in data initialization stage [class CustomizationDataset](https://github.com/THUDM/GLM/blob/73bdad08b04f3391a0710ec8ac3968d2db327f56/tasks/seq2seq/dataset.py#L779),  the source tokens are padding to the length of `max_src_length` (line 834) and I find that the `attention_mask` is set to the length of  source tokens with padding tokens (line 858). See below code snippet.
``` python
        # tasks/seq2seq/dataset.py
        ## line 834
        if len(source_tokens) < self.max_src_length:
            source_tokens = source_tokens + [pad_id] * (self.max_src_length - len(source_tokens))
        sep = len(source_tokens)
        ## line 858
        sample = {'text': np.array(tokens, dtype=np.int64), 'target': np.array(target_ids, dtype=np.int64),
                  'attention_mask': np.array(sep, dtype=np.int64),
                  'loss_mask': np.array(loss_mask, dtype=np.int64),
                  "position_id": np.array(position_ids, dtype=np.int64), "uid": example.guid}
```
So, this leads to that the attention mask value of padding token is set to `1`, and thus source token can 'see' the padding token. 

Is it correct? Maybe the code around line 834 should be changed to the below code to make padding token invisible to source tokens?
``` python
        # tasks/seq2seq/dataset.py
        ## line 834
        sep = len(source_tokens)
        if len(source_tokens) < self.max_src_length:
            source_tokens = source_tokens + [pad_id] * (self.max_src_length - len(source_tokens))
```

Thanks!

**Comment by duzx16** [comment]
This strategy is the same as that in our seq2seq experiments. Empirically we found that the model learned to ignore the padding tokens with a specific id after finetuning.

**Comment by duzx16** [comment]
Now you can add `--mask-pad-token` argument to set the attention mask of padding to 0.

**Comment by Bo396543018** [comment]
Hi, i found that the padding mask for the target part was not set to 0. Is it more reasonable to set it to 0 during padding?
